modules/nf-core/bedtools/unionbedg/main.nf

```diff
@@ -0,0 +1,42 @@
+process BEDTOOLS_UNIONBEDG {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::bedtools=2.30.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bedtools:2.30.0--h7d7f7ad_2':
+        'quay.io/biocontainers/bedtools:2.30.0--h7d7f7ad_2' }"
+
+    input:
+    tuple val(meta), path(bedgraph, stageAs: "?/*")
+    path chrom_sizes
+    val(empty)
+
+    output:
+    tuple val(meta), path("*.bed"), emit: bed
+    path  "versions.yml"          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def sizes = chrom_sizes ? "-g ${chrom_sizes}" : ''
+    def empty = empty ? "-empty" : ''
+    if ("$bedgraph" == "${prefix}.bed") error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
```

Comment:
I think it would be better to add a loop here to check every file separately. Here it checks all files in a string (which will never be the same as the output file)

Reply:
Or can I just remove this "if" statement?

Reply:
I wouldn't do that, since this will give a warning before the module has been run, saving some precious compute time in the case of an error :)

Reply:
So how should I loop?

Reply:
I think something like this should work:

```groovy
bedgraph.collect { if ("$it" == "${prefix}.bed") error "$it has the same name as the output, use \"task.ext.prefix\" to disambiguate!" }
```

Either that or with `.each { }` instead of `.collect { }`. I don't remember which one it was :)